src/test/compile-fail/issue-37665.rs

```diff
@@ -0,0 +1,20 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![feature(rustc_attrs)]
+
+use std::path::MAIN_SEPARATOR;
+
+#[rustc_error]
+fn main() { //~ ERROR compilation successful
+    let mut foo : String = "hello".to_string();
+    foo.push(MAIN_SEPARATOR);
+    println!("{}", foo);
+}
```

Comment:
I believe this test needs to have `--unpretty mir -Z unstable-options` passed to the compiler. `// compile-flags: -Z unstable-options --unpretty=mir` should work at the top of the file.